File: src/main/java/fr/tikione/c2e/service/web/scrap/CPCScraperService.java
package fr.tikione.c2e.service.web.scrap;

import fr.tikione.c2e.model.web.Article;
import org.jsoup.nodes.Document;

import java.util.List;
import java.util.Map;

public interface CPCScraperService {
    
    /** Extract web page articles and associate this operation to a general relevance score. */
    Map<Integer, List<Article>> extractArticles(Document doc);
    
    List<Article> extractSingleNews(Document doc);
    
    List<Article> extractNews(Document doc);
    
    List<Article> extractTests(Document doc);
    
    List<Article> extractShortTests(Document doc);
    
    List<Article> extractPlumePudding(Document doc);
    
    List<Article> extractComing(Document doc);
    
    List<Article> extractUnderConstruction(Document doc);
    
    List<Article> extractTechno(Document doc);
    
    List<Article> extractStudy(Document doc);
    
    List<Article> extractEverythingElse(Document doc);
}

File: src/main/kotlin/fr/tikione/c2e/service/web/scrap/CPCScraperService.kt
package fr.tikione.c2e.service.web.scrap

import fr.tikione.c2e.model.web.Article
import org.jsoup.nodes.Document

interface CPCScraperService {

    /** Extract web page articles and associate this operation to a general relevance score.  */
    fun extractArticles(doc: Document): Map<Int, List<Article>>

    fun extractSingleNews(doc: Document): List<Article>

    fun extractNews(doc: Document): List<Article>

    fun extractTests(doc: Document): List<Article>

    fun extractShortTests(doc: Document): List<Article>

    fun extractPlumePudding(doc: Document): List<Article>

    fun extractComing(doc: Document): List<Article>

    fun extractUnderConstruction(doc: Document): List<Article>

    fun extractTechno(doc: Document): List<Article>

    fun extractStudy(doc: Document): List<Article>

    fun extractEverythingElse(doc: Document): List<Article>
}
